pkg/controller/machinepool/azureactuator.go

```diff
@@ -85,13 +85,39 @@ func (a *AzureActuator) GenerateMachineSets(cd *hivev1.ClusterDeployment, pool *
 		}
 	}
 
+	var hyperVGen string
 	if osImage := pool.Spec.Platform.Azure.OSImage; osImage.Publisher != "" {
 		computePool.Platform.Azure.OSImage = installertypesazure.OSImage{
 			Publisher: osImage.Publisher,
 			Offer:     osImage.Offer,
 			SKU:       osImage.SKU,
 			Version:   osImage.Version,
 		}
+	} else {
+		// An image was not provided so check if the installer created a "gen2" image
+		// to determine if we should allow resultant machinesets to consume a "gen2" image.
+		gen2ImageExists, err := a.gen2ImageExists(cd.Spec.ClusterMetadata.InfraID, ic.Platform.Azure.ClusterResourceGroupName(cd.Spec.ClusterMetadata.InfraID))
+		if err != nil {
+			return nil, false, err
+		}
+
+		// Get HyperV Generation to provide to installer API. The HyperV Generation is germane to the instance/disk
+		// type and affects the image used for the instance. "-gen-2" will be appended to the image name when the
+		// hyperVGen == "V2".
+		hyperVGen, err = a.client.GetHyperVGenerationVersion(context.TODO(),
+			computePool.Platform.Azure.InstanceType,
+			computePool.Platform.Azure.OSDisk.DiskType,
+			cd.Spec.Platform.Azure.Region,
+		)
+		if err != nil {
+			return nil, false, err
+		}
```

Comment:
Does this chunk need to be done if `!gen2ImageExists`?

Reply:
No, if `!gen2ImageExists` then we do not need to determine if the machine in question can support `"V2"`.

Reply:
Updated to not get the hyperVGen when `!gen2ImageExists`